packages/js/block-templates/src/utils/register-woo-block-type.ts

```diff
@@ -66,11 +68,32 @@ function getEdit<
 			[ getEvaluationContext, hideConditions ]
 		);
 
+		const disabled: boolean = useSelect(
+			( select: typeof WPSelect ) => {
+				if (
+					! disableConditions ||
+					! Array.isArray( disableConditions )
+				) {
+					return false;
+				}
+
+				const evaluationContext = getEvaluationContext( select );
+
+				return disableConditions.some( ( condition ) =>
+					evaluate( condition.expression, evaluationContext )
+				);
+			},
+			[ getEvaluationContext, disableConditions ]
+		);
```

Comment:
I think that it is recommended to combine things in a single `useSelect` when possible. In this case, it is and I think we should combine this useSelect with the one above.